drafts/2019-05-16-restful-config-viewing-and-validation-service.adoc

```diff
@@ -0,0 +1,327 @@
+
+```json
+[
+   {
+      "configElementName": "connectionFactory",
+      "uid": "cf1",
+      "id": "cf1",
+      "jndiName": "eis/cf1",
+      "containerAuthDataRef": [
+         {
+            "configElementName": "containerAuthData",
+            "uid": "connectionFactory[cf1]/containerAuthData[default-0]",
+            "password": "******",
+            "user": "containerAuthUser1"
+         }
+      ],
+      "properties.TestValidationAdapter.ConnectionFactory": [
+         {
+            "hostName": "myhost.openliberty.io",
+            "password": "******",
+            "portNumber": 9876,
+            "userName": "DefaultUserName"
+         }
+      ]
+   }
+]
+```
+
+To view an individual connection factory append the `uid`. In our case this will be https://localhost:9443/ibm/api/config/connectionFactory/cf1.
+
+```json
+{
+   "configElementName": "connectionFactory",
+   "uid": "cf1",
+   "id": "cf1",
+   "jndiName": "eis/cf1",
+   "containerAuthDataRef": [
+      {
+         "configElementName": "containerAuthData",
+         "uid": "connectionFactory[cf1]/containerAuthData[default-0]",
+         "password": "******",
+         "user": "containerAuthUser1"
+      }
+   ],
+   "properties.TestValidationAdapter.ConnectionFactory": [
+      {
+         "hostName": "myhost.openliberty.io",
+         "password": "******",
+         "portNumber": 9876,
+         "userName": "DefaultUserName"
+      }
+   ]
+}
+```
+
+== Validating a JCA Connection Factory
+
+The REST endpoints for validating a connection factory can be found at https://localhost:9443/ibm/api/validation/connectionFactory/{uid}. To test `cf1` using container authentication we can use the following: https://localhost:9443/ibm/api/validation/connectionFactory/cf1?auth=container
+
+```json
+{
+   "uid": "cf1",
+   "id": "cf1",
+   "jndiName": "eis/cf1",
+   "successful": true,
+   "info": {
+      "resourceAdapterName": "TestValidationAdapter",
+      "resourceAdapterVersion": "28.45.53",
+      "resourceAdapterJCASupport": "1.7",
+      "resourceAdapterVendor": "OpenLiberty",
+      "resourceAdapterDescription": "This tiny resource adapter doesn't do much at all.",
+      "eisProductName": "TestValidationEIS",
+      "eisProductVersion": "33.56.65",
+      "user": "containerAuthUser1"
+   }
+}
+```
+
+Validation of a connection factory supports both container and application authentication via the `auth` parameter being included on the URL.  Additionally, when using `?auth=application`, a user can be specified by including the `X-Validation-User` and `X-Validation-Password` headers. Finally, the authentication alias can be specified using the `authAlias` parameter. For example, this could look like https://localhost:9443/ibm/api/validation/connectionFactory/cf1?auth=container&authAlias=auth2.
+
+== Cloudant Database
+
+Cloudant databases can be viewed and validated as well.
+
+Lets use the following server config snippets:
+
+```xml
+...
```

Comment:
I think we should still include the open/close `<server>` tags here, and also the `<featureManager>` tags, so the reader has the proper context on where this XML is supposed to go

Reply:
Done, made it a full server.xml since it only needed 3 more lines.